content/src/view.js

```diff
@@ -3172,6 +3380,147 @@ function setupHpanelBox(box) {
+    if(arrow_data["before"]){
+      firstBeforeLoc = arrow_data['before']["first"];
+      secondBeforeLoc = arrow_data['before']['second'];
+    }
+    if(arrow_data["after"]){
+      firstAfterLoc = arrow_data["after"]["first"];
+      secondAfterLoc = arrow_data["after"]["second"];
+    }
+  }
+          
+  if (firstBeforeLoc.first_line != undefined && secondBeforeLoc.first_line != undefined){
+    
+    if (block_mode){
+      var dropletEditor = state.pane[pane].dropletEditor;
+      var startBoundsBefore = dropletEditor.getLineMetrics(firstBeforeLoc.first_line);
+      var endBoundsBefore = dropletEditor.getLineMetrics(secondBeforeLoc.first_line);
+      startcoordsBefore = {pageX : startBoundsBefore.bounds.x, pageY: startBoundsBefore.bounds.y};
+      endcoordsBefore =  {pageX : endBoundsBefore.bounds.x, pageY: endBoundsBefore.bounds.y};
+      offset_top_before = startBoundsBefore.bounds.height - 30;
+      offset_left_before = Math.max(startBoundsBefore.bounds.width, endBoundsBefore.bounds.width)  + 20;
+
+    }
+    else{
+      offset_top_before = $(".editor").offset().top ;
+      offset_left_before = $(".editor").offset().left + 30;
+      startcoordsBefore = state.pane[pane].editor.renderer.textToScreenCoordinates((firstBeforeLoc.first_line), (firstBeforeLoc.last_column + 10));
+      endcoordsBefore = state.pane[pane].editor.renderer.textToScreenCoordinates((secondBeforeLoc.first_line ), (secondBeforeLoc.last_column + 10));
+    }
+    if (Math.abs(secondBeforeLoc.first_line - firstBeforeLoc.first_line) > 1){
+
+      var x_val = 0;
+      if(startcoordsBefore.pageX > endcoordsBefore.pageX){
+        x_val = startcoordsBefore.pageX;
+      } 
+      else{
+        x_val = endcoordsBefore.pageX;
+      }
+    }
+    if (show_fade){
+      before_arrowtext = "<path d='" + curvedVertical(x_val + offset_left_before, (startcoordsBefore.pageY - offset_top_before), x_val + offset_left_before, (endcoordsBefore.pageY - offset_top_before)) 
+                        + "' marker-start='url(#arrowhead2)' style='stroke:#8EC8FF; fill:none;' position='relative'/> \ "
+    }
+    else{
+     before_arrowtext = "<path d='" + curvedVertical(x_val + offset_left_before, (startcoordsBefore.pageY - offset_top_before), x_val + offset_left_before, (endcoordsBefore.pageY - offset_top_before)) 
+                        + "' marker-start='url(#arrowhead3)' style='stroke:dodgerblue; fill:none;' position='relative'/> \ "
+    } 
+  }
+  if (firstAfterLoc.first_line != undefined && secondAfterLoc.first_line != undefined){
+    if (block_mode){
+      var startBoundsAfter= dropletEditor.getLineMetrics(firstAfterLoc.first_line);
+      var endBoundsAfter = dropletEditor.getLineMetrics(secondAfterLoc.first_line);
+      startcoordsAfter = {pageX : startBoundsAfter.bounds.x, pageY: startBoundsAfter.bounds.y};
+      endcoordsAfter =  {pageX : endBoundsAfter.bounds.x, pageY: endBoundsAfter.bounds.y};
+      offset_top_after = startBoundsAfter.bounds.height - 30;
```

Comment:
What's the 30 pixels from?  What if the font size is smaller?

Your arrow looks like it starts "after" (below) the current line and goes up to maybe 30 pixels above the top - not sure.  But to my eyes, it might look better if it came from the "middle" of the current line instead of below it.